docs/SharedBrokerPool.md

```diff
@@ -0,0 +1,43 @@
+---
+id: shared-broker-pool
+title: Sharing Broker Pools
+---
+
+The default behavior of KafkaJS is to treat each consumer, producer, and admin client independently from each other. In particular this means network connections to the configured brokers will be created again for every one of these. For larger applications this can lead to high resource usage both on the client side as well as on the broker side, and it may lead to throttling.
+
+It is possible to influence this behavior for consumers and producers by manually configuring "broker pools", and sharing these.
+
+## <a name="consumer"></a> Example
+
+```javascript
+const brokerPool = kafka.brokerPool({ })
+await brokerPool.connect()
+```
+
+This broker pool can be provided to the `kafka.consumer` and `kafka.producer` methods, instead of the options that would typically configure the pool.
```

Comment:
Any reason not to do the same for the admin client? Granted, it is typically not used for long running processes where this would really matter, but it could be, and as long as it doesn't make the admin client more complicated, I don't see any reason not to allow it.

Reply:
Allowing is fine, the problem was mainly that the `admin()` function would always force the `allowTopicCreation` setting, and I didn't want to spend too much time thinking about the implications.
But yes, I think we could simply allow that as well, and make the caller responsible for knowing what they do -- which I think is fair for an "advanced" use case.

Reply:
I think the way forward here is probably to not just check whether you provide any settings at all, but rather expose them on the connection pool entity so that the `admin`/`consumer`/`producer` functions can easily check for "compatibility".

I'll hopefully get to do that tomorrow as well, which might help further with making the compatibility check more obvious.